.github/workflows/spell-check.yml

```diff
@@ -0,0 +1,68 @@
+name: Spell Check
+
+on:
+  # Run on-demand using workflow_dispatch
+  workflow_dispatch:
+  # Schedule to run once a week
+  schedule:
+    - cron: '0 0 * * 1' # Runs every Monday at midnight UTC
+
+jobs:
+  spelling:
+    name: Spell check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install spell checker
+        run: |
+          pip install check-spelling
+
+      - name: Run spell checker
+        uses: check-spelling/check-spelling@master
```

Comment:
Please use a specific version here instead of depending on the master. Once done we will review the particular version during approval process.